doc/hardware/emulator/gpio_emulator.rst

```diff
@@ -0,0 +1,139 @@
+.. _gpio_emul:
+
+GPIO Emulator
+#############
+
+Overview
+========
+
+Zephyr's GPIO (General Purpose Input-Output) emulator has several functions.
+
+1. It serves as a model for the basic GPIO Driver API; if a new GPIO driver is added for a new
+    controller, and some aspect of the new driver fails Zephyr's GPIO Driver API Testsuite, while
+    the emulated GPIO driver (and others) pass, it may be an indicator that something in the new
+    driver does not follow the same behaviour required by Zephyr's GPIO Driver Model, and the new
+    driver requires improvement.
+
+2. The ``gpio_emul`` device allows us to perform fast functional testing to verify the expected
+    behaviour of various libraries and applications. Why? In Zephyr, a GPIO is a GPIO is a GPIO
+    (from a functional perspective) because there is a common GPIO Driver API and model. How can
+    emulated GPIO be used in an application? Well, the GPIO Emulator also has programmable
+    behaviour via the backend API. So the application can inject custom input events, create
+    custom "wiring" via the existing GPIO Driver API.
+
+3. Lastly, ``gpio_emul`` is extensible, which means it can be repurposed where other, virtual
+    GPIO might be used. For example, the SDL Button driver.
+
+
+Usage
+=====
+
+Use the emulated GPIO driver exactly the same as one would use any other GPIO driver.
```

Comment:
Related to the above discussion of formality in documentation, "Use the emulated GPIO driver" is fairly informal, while "one would use" is fairly formal. Without expressing a strong opinion on which one I think you should choose, this sentence and the document should be internally consistent: "Use the emulated GPIO driver ... as you would use ..." or "One uses the emulated GPIO driver ... as one would use ..."

The same idea also applies to other constructions below like "may be specified" ("you may specify") and "it may be necessary" ("you may need").